Go 1.15: conversion from int to string yields a string of one rune

**What did you do?**

go test

**What did you expect to see?**

tests pass.

**What did you see instead? Under which circumstances?**

```
ok  	github.com/prometheus/prometheus/tsdb/fileutil	0.007s
github.com/prometheus/prometheus/tsdb/index
# github.com/prometheus/prometheus/tsdb/index
./index_test.go:521:21: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
./index_test.go:535:22: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
./index_test.go:541:29: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
./index_test.go:545:27: conversion from untyped int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
./index_test.go:551:22: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
FAIL	github.com/prometheus/prometheus/tsdb/index [build failed]
```

**Environment**

Go 1.15 rc 1 on Fedora Linux Rawhide

* System information:

	various system (x86_64, armv7hl, i686…)

* Prometheus version:

	2.20.0

* Alertmanager version:

	0.20.0


Please refer to https://github.com/golang/go/issues/32479

--- a/tsdb/index/index_test.go
+++ b/tsdb/index/index_test.go
@@ -523,7 +523,7 @@ func TestSymbols(t *testing.T) {
 	buf.PutBE32int(100) // Number of symbols.
 	for i := 0; i < 100; i++ {
 		// i represents index in unicode characters table.
-		buf.PutUvarintStr(string(i)) // Symbol.
+		buf.PutUvarintStr(string(rune(i))) // Symbol.
 	}
 	checksum := crc32.Checksum(buf.Get()[symbolsStart+4:], castagnoliTable)
 	buf.PutBE32(checksum) // Check sum at the end.
@@ -537,23 +537,23 @@ func TestSymbols(t *testing.T) {
 	for i := 99; i >= 0; i-- {
 		s, err := s.Lookup(uint32(i))
 		testutil.Ok(t, err)
-		testutil.Equals(t, string(i), s)
+		testutil.Equals(t, string(rune(i)), s)
 	}
 	_, err = s.Lookup(100)
 	testutil.NotOk(t, err)
 
 	for i := 99; i >= 0; i-- {
-		r, err := s.ReverseLookup(string(i))
+		r, err := s.ReverseLookup(string(rune(i)))
 		testutil.Ok(t, err)
 		testutil.Equals(t, uint32(i), r)
 	}
-	_, err = s.ReverseLookup(string(100))
+	_, err = s.ReverseLookup(string(rune(100)))
 	testutil.NotOk(t, err)
 
 	iter := s.Iter()
 	i := 0
 	for iter.Next() {
-		testutil.Equals(t, string(i), iter.At())
+		testutil.Equals(t, string(rune(i)), iter.At())
 		i++
 	}
 	testutil.Ok(t, iter.Err())

